src/traces/scatterternary/plot.js

```diff
@@ -13,11 +13,16 @@ var scatterPlot = require('../scatter/plot');
 
 
 module.exports = function plot(ternary, data) {
+    var plotContainer = ternary.plotContainer;
+
+    // remove all nodes inside the scatter layer
+    plotContainer.select('.scatterlayer').selectAll('*').remove();
```

Comment:
I'm not sure how d3 manages selections internally, but I'd guess `<selection>.html('')` would be faster than `selectAll('*').remove()`?